### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by Cerdtruir**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -75,6 +116,22 @@ def add_teacher(name, age)
...
+    save_data_array = []
+    @people.each do |person|
+      save_person = {}
+      save_person['type'] = person.class.name
+      person.instance_variables.map do |attribute|
+        save_person[attribute] = person.instance_variable_get(attribute)
+      end
+      person.rentals.instance_variables.map do |attribute|
+        save_person.rentals[attribute] = person.instance_variable_get(attribute)
+      end
```
Comment (ikraamg) [inline app.rb]: This does not seem to do anything. 
Also, this is iterating over the instance variables on the **array of rentals** and not the rentals themselves.

File: app.rb [hunk 2]
```diff
@@ -75,6 +116,22 @@ def add_teacher(name, age)
...
   end
 
+  def save_people
+    save_data_array = []
+    @people.each do |person|
+      save_person = {}
+      save_person['type'] = person.class.name
+      person.instance_variables.map do |attribute|
+        save_person[attribute] = person.instance_variable_get(attribute)
+      end
```
Comment (ikraamg) [inline app.rb]: It is cool to loop over the instance_variables but rather use select the variables that you want to store. It does not makes sense to store the rentals, nor the Corrector class instance string in the JSON file:


{"type":"Teacher","@id":886,"@name":"teacher","@age":"55","@parent_permission":true,"@corrector":"#<Corrector:0x00007fd9c10b67b8>",,"@specialization":"apples"}]

File: app.rb [hunk 3]
```diff
@@ -83,11 +140,23 @@ def add_book
...
 
+  def save_books
+    save_data_array = []
+    @books.each do |book|
+      save_book = {}
+      book.instance_variables.map do |attribute|
+        save_book[attribute] = book.instance_variable_get(attribute)
+      end
+      save_data_array << save_book
+    end
```
Comment (ikraamg) [inline app.rb]: This also produces an unnecessary "@rentals" attribute in the json.

[{"@title":"speaks","@author":"you","@rentals":[]}]

**Review by ikraamg — COMMENTED** [review]
Looks good overall 👍 

Left some comments on cleaning up the implementation.

**Comment by Cerdtruir** [comment]
Yeah, i see its much simpler to get the data i need instead

**Pull request merged**